OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -649,20 +649,14 @@ public static DreamValue NativeProc_fcopy_rsc(NativeProc.Bundle bundle, DreamObj
         public static DreamValue NativeProc_fdel(NativeProc.Bundle bundle, DreamObject? src, DreamObject? usr) {
             DreamValue file = bundle.GetArgument(0, "File");
 
-            string filePath;
+            string? filePath;
             if (file.TryGetValueAsDreamResource(out var resource)) {
                 filePath = resource.ResourcePath;
             } else if(!file.TryGetValueAsString(out filePath)) {
                 throw new Exception($"{file} is not a valid file");
             }
 
-            bool successful;
-            if (filePath.EndsWith("/")) {
-                successful = bundle.ResourceManager.DeleteDirectory(filePath);
-            } else {
-                successful = bundle.ResourceManager.DeleteFile(filePath);
-            }
-
+            bool successful = filePath!.EndsWith("/") ? bundle.ResourceManager.DeleteDirectory(filePath) : bundle.ResourceManager.DeleteFile(filePath);
```

Comment:
This is still suppressing the issue I pointed out earlier. `resource.ResourcePath` could be null, which is why the compiler warned about it in the first place.